bundles/org.openhab.binding.etherrain/src/main/java/org/openhab/binding/etherrain/internal/handler/EtherRainHandler.java

```diff
@@ -0,0 +1,240 @@
+ */
+public class EtherRainHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(EtherRainHandler.class);
+
+    private EtherRainCommunication device = null;
+    private boolean connected = false;
+    EtherRainConfiguration config;
+
+    private ScheduledFuture<?> updateJob;
+
+    private HttpClient httpClient;
+
+    /*
+     * Constructor class. Only call the parent constructor
+     */
+    public EtherRainHandler(Thing thing, HttpClient httpClient) {
+        super(thing);
+        this.httpClient = httpClient;
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (command.toString().equals("REFRESH")) {
+            updateBridge();
+            return;
+        }
+
+        else if (channelUID.getId().equals(EtherRainBindingConstants.CHANNEL_ID_EXECUTE)) {
+            execute();
+            updateState(EtherRainBindingConstants.CHANNEL_ID_EXECUTE, OnOffType.OFF);
+        }
+
+        else if (channelUID.getId().equals(EtherRainBindingConstants.CHANNEL_ID_CLEAR)) {
+            clear();
+            updateState(EtherRainBindingConstants.CHANNEL_ID_CLEAR, OnOffType.OFF);
+        }
+    }
+
+    private boolean connectBridge() {
+        config = getConfigAs(EtherRainConfiguration.class);
+
+        logger.debug(
+                "Attempting to connect to Etherrain with config = (Host: {}, Port: {}, Password: {}, Refresh: {}).",
+                config.host, config.port, config.password, config.refresh);
+
+        device = new EtherRainCommunication(config.host, config.port, config.password, httpClient);
+
+        EtherRainStatusResponse response;
+        try {
+            response = device.commandStatus();
+        } catch (EtherRainException | IOException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR,
+                    "Could not create a connection to the EtherRain");
+            logger.debug("Could not open API connection to the EtherRain device. Exception received: {}", e.toString());
+            device = null;
+            updateStatus(ThingStatus.OFFLINE);
+            return false;
+        }
+        if (response == null) {
+            logger.debug("Command Status returned null");
+            device = null;
+            updateStatus(ThingStatus.OFFLINE);
+            return false;
+        }
+
+        updateStatus(ThingStatus.ONLINE);
+
+        return true;
+    }
+
+    private void startUpdateJob() {
+        logger.debug("Starting Etherrain Update Job");
+        updateJob = scheduler.scheduleWithFixedDelay(this::updateBridge, 0, config.refresh, TimeUnit.SECONDS);
+
+        logger.debug("EtherRain sucessfully initialized. Starting status poll at: {}", config.refresh);
+    }
+
+    private void stopUpdateJob() {
+        logger.debug("Stopping Etherrain Update Job");
+        if (updateJob != null && !updateJob.isDone()) {
```

Comment:
```suggestion
        final updateJob = this.updateJob;
        if (updateJob != null && !updateJob.isDone()) {
```

Reply:
I am assuming you meant:

 final ScheduledFuture<?> updateJob = this.updateJob;

Reply:
Sure.